src/components/Dashboard/DashboardDevice.tsx

```diff
@@ -0,0 +1,231 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import React, { useMemo } from 'react';
+import { CompositeFeature } from 'types';
+import { BaseFeatureProps } from 'components/features/base';
+import DeviceFooter from 'components/Dashboard/DeviceFooter';
+
+import styles from 'components/Dashboard/DashboardDevice.scss';
+import ReactSwitch from 'react-switch';
+
+type Props = BaseFeatureProps<CompositeFeature>;
+
+type State = {
+    isThermostat?: true;
+    isSocket?: true;
+    hasTemperature?: true;
+    hasHumidity?: true;
+    hasContact?: true;
+    hasSwitch?: true;
+    hasOccupancy?: true;
+    hasWaterLeak?: true;
+    hasClick?: true;
+};
+
+const DashboardDevice: React.FC<Props> = (props) => {
+    const state = useMemo(() => {
+        const state: Partial<State> = {};
+        if (Array.isArray(props.feature.features)) {
```

Comment:
you don't need to check whether its array or not, perhaps you might want to use `isCompositeFeature` type guard

Reply:
`isCompositeFeature` is only type guard, but will not solve the issue that features are not there always it's much better to check it before usage or ofc can use nullish but do it properly :)

Reply:
I know, but if feature is composite but features filed is null(not array) this means something is wrong with device definition, in this case we will crash and user will go to GitHub to raise issue and we will fix the problem source (not symptoms)

Reply:
okay then from
```
  <Route
      path="/device/:dev/:tab?"
      render={(props) => (
          <ErrorBoundary {...props}>
              <ConnectedDevicePage />
          </ErrorBoundary>
      )}
  />
```
  change it to
  ```
   <Route
      path="/device/:dev/:tab?"
      render={(props) => (
          <ErrorBoundary>
              <ConnectedDevicePage {...props} />
          </ErrorBoundary>
      )}
  />
  ```
and it will make more sense for sure, what do you think?

Reply:
Yes, this is fine, but please test carefully that error boundary works correct after this change (you can navigate to page witch any error and then navigate back to any page)